### PR TITLE
feat: Option to resend verification mail

### LINKED ISSUE #1472: Add option to resend verification mail
**Describe the feature you'd like**
User should have an option to trigger resending of verification mail for unverified accounts.

**Would you like to work on the issue?**
- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by ShridharGoel**
Fixes #1472 

Checklist:

- [ ] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Changes: 
Now, the user would have the option to trigger resending of verification mail for an unverified account.

GIF for the change:
ezgif com-video-to-gif 23

### CODE REVIEW

File: app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailPresenter.java [hunk 1]
```diff
@@ -34,10 +43,21 @@ public void loadOrganizer(boolean forceReload) {
             .compose(progressiveErroneousRefresh(getView(), forceReload))
             .subscribe(loadedUser -> {
                 this.user = loadedUser;
+                resendVerificationMail
+                    .setEmail(user.email);
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailPresenter.java]: Use user.getEmail()

File: app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailPresenter.java [hunk 2]
```diff
@@ -34,10 +43,21 @@ public void loadOrganizer(boolean forceReload) {
...
             }, Logger::logError);
     }
 
+    public void resendVerificationMail() {
+        compositeDisposable.add(
+            resendVerificationMailModel.resendVerificationMail(resendVerificationMail)
+                .doOnSubscribe(disposable -> getView().showProgress(true))
+                .doFinally(() -> getView().showProgress(false))
+                .subscribe(resendMailResponse -> getView().showSnackbar("Verification Mail Resent"),
+                    throwable -> getView().showError(ErrorUtils.getMessage(throwable).toString())));
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailPresenter.java]: Handle error correctly, don't just show to the user

File: app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailView.java [hunk 3]
```diff
@@ -6,4 +6,12 @@
 import com.eventyay.organizer.common.mvp.view.Refreshable;
 import com.eventyay.organizer.data.user.User;
 
-public interface OrganizerDetailView extends Progressive, Erroneous, Refreshable, ItemResult<User> { }
+public interface OrganizerDetailView extends Progressive, Erroneous, Refreshable, ItemResult<User> {
+    @Override
+    void showProgress(boolean show);
+
+    @Override
+    void showError(String error);
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailView.java]: Both of the above methods arent needed. They are already there

Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailView.java]: Removed

File: app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailPresenter.java (hunk 2 above)
Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailPresenter.java]: Is it okay now?

File: app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailPresenter.java (hunk 1 above)
Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailPresenter.java]: Done

File: app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailPresenter.java [hunk 4]
```diff
@@ -54,7 +55,10 @@ public void resendVerificationMail() {
                 .doOnSubscribe(disposable -> getView().showProgress(true))
                 .doFinally(() -> getView().showProgress(false))
                 .subscribe(resendMailResponse -> getView().showSnackbar("Verification Mail Resent"),
-                    throwable -> getView().showError(ErrorUtils.getMessage(throwable).toString())));
+                    throwable -> {
+                        getView().showError(ErrorUtils.getErrorDetails(throwable).toString());
+                        Timber.e(throwable, "An exception occurred : %s", throwable.getMessage());
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailPresenter.java]: Still not handling anything

Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailPresenter.java]: Can you please explain a bit about what change is required?

**Pull request merged**